airflow/executors/base_executor.py

```diff
@@ -376,6 +376,18 @@ def terminate(self):
         """This method is called when the daemon receives a SIGTERM."""
         raise NotImplementedError()
 
+    def cleanup_stuck_queued_tasks(self, tis: list[TaskInstance]) -> list[str]:  # pragma: no cover
+        """
+        Handle remnants of tasks that were failed because they were stuck in queued.
+        Tasks can get stuck in queued. If such a task is detected, it will be marked
+        as `UP_FOR_RETRY` if the task instance has remaining retries or marked as `FAILED`
+        if it doesn't.
+
+        :param tis: List of Task Instances to clean up
+        :return: List of readable task instances for a warning message
+        """
+        raise NotImplementedError()
```

Comment:
I just realized we haven't properly handled what happens when another executor hits this (however unlikely).

Easy way to reproduce this would be setting a really low task_queued_timeout, and add a longer sleep to the top level of a DAG, when using LocalExecutor.

Maybe log this should just call fail also? It shouldn't kill the scheduler I'd think.